src/actions/articles.js

```diff
@@ -0,0 +1,161 @@
+import { articlesType } from '../actionTypes';
+import * as helper from '../helpers';
+
+export const getAllArticles = () => async (dispatch) => {
+  helper
+    .axios()
+    .get('/articles')
+    .then(({ data }) => {
+      dispatch({
+        type: articlesType.FETCH_ARTICLES,
+        payload: data.articles
+      });
+      return data;
+    })
+    .catch((error) => {
+      if (error) {
+        dispatch({
+          type: articlesType.ERROR_ARTICLE,
+          payload: error.response.data
+        });
+        return error;
+      }
+      return true;
+    });
+};
+
+export const fetchOneArticle = slug => async (dispatch) => {
+  await helper
+    .axios()
+    .get(`/articles/${slug}`)
+    .then(({ data }) => {
+      dispatch({
+        type: articlesType.FETCH_ARTICLE,
+        payload: data.article
+      });
+      return data;
+    })
+    .catch((error) => {
+      if (error) {
+        dispatch({
+          type: articlesType.ERROR_ARTICLE,
+          payload: error.response.data
+        });
+        return error;
+      }
+      return true;
+    });
+};
+
+export const createPost = postData => async (dispatch) => {
+  const uploadConfig = { headers: { 'content-type': 'multipart/form-data' } };
+  await helper
+    .axios()
+    .post('/articles', postData, uploadConfig)
+    .then(({ data }) => {
+      dispatch({
+        type: articlesType.CREATE_ARTICLE,
+        payload: data.article
+      });
+      return data;
+    })
+    .catch((error) => {
+      console.log('!@$', error.response);
+      if (error.response.data.errors) {
+        dispatch({
+          type: articlesType.ERROR_ARTICLE,
+          payload: error.response.data.errors
+        });
+        return error;
+      }
+      return true;
+    });
+};
+
+export const editPost = (data = {}) => async (dispatch) => {
+  const post = {
+    title: data.article.title,
+    description: data.article.description,
+    body: data.article.body
+  };
+  await helper
+    .axios()
+    .put(`/articles/${data.slug}`, post)
+    .then(({ data }) => {
```

Comment:
Let use Async and await instead of Then promises.

Reply:
Noted, thanks